### PR TITLE
Natasha/categories feed

### PR DESCRIPTION
**Pull request by natashaarmbrust**
I added expandable cells to the table view for episodes with long descriptions. I'll ask @dantheli at next meeting if theres any cool animations we can do with this.

### CODE REVIEW

File: Podcast/DiscoverTableViewCell.swift [hunk 1]
```diff
@@ -125,6 +128,10 @@ class DiscoverTableViewCell: UITableViewCell {
         contentView.addSubview(clickToPlayImageButton)
         
         adjustForScreenSize()
+        
+        heightConstraint = NSLayoutConstraint(item: self.contentView, attribute: NSLayoutAttribute.height, relatedBy: NSLayoutRelation.equal, toItem: nil, attribute: NSLayoutAttribute.notAnAttribute, multiplier: 1, constant: height)
+    
+        self.contentView.addConstraint(heightConstraint!)
```
Comment (anniexcheng) [inline Podcast/DiscoverTableViewCell.swift]: lines 132 and 134: no `self`

File: Podcast/DiscoverTableViewCell.swift [hunk 2]
```diff
@@ -125,6 +128,10 @@ class DiscoverTableViewCell: UITableViewCell {
         contentView.addSubview(clickToPlayImageButton)
         
         adjustForScreenSize()
+        
+        heightConstraint = NSLayoutConstraint(item: self.contentView, attribute: NSLayoutAttribute.height, relatedBy: NSLayoutRelation.equal, toItem: nil, attribute: NSLayoutAttribute.notAnAttribute, multiplier: 1, constant: height)
```
Comment (anniexcheng) [inline Podcast/DiscoverTableViewCell.swift]: no `NSLayoutAttribute` and `NSLayoutRelation`

File: Podcast/DiscoverTableViewCell.swift [hunk 3]
```diff
@@ -155,7 +175,8 @@ class DiscoverTableViewCell: UITableViewCell {
         episodeDateLabel.frame = CGRect(x: seriesNameLabel.frame.maxX, y: seriesNameLabelMinY, width: 0, height: 0)
         episodeDateLabel.sizeToFit()
         
-        seperator.frame = CGRect(x: 0, y: height - seperatorHeight, width: self.frame.width, height: seperatorHeight)
+        seperator.frame = CGRect(x: 0, y: self.frame.height - seperatorHeight, width: self.frame.width, height: seperatorHeight)
```
Comment (anniexcheng) [inline Podcast/DiscoverTableViewCell.swift]: no `self`

File: Podcast/DiscoverViewController.swift [hunk 4]
```diff
@@ -47,13 +47,15 @@ class DiscoverViewController: UIViewController, UITableViewDelegate, UITableView
         categoryCollectionView.reloadData()
         
         //tableview
-        feedTableView = UITableView(frame: CGRect(x: 0, y: topView.frame.height, width: self.view.frame.width, height: self.view.frame.height - topView.frame.height))
+        feedTableView = UITableView(frame: CGRect(x: 0, y: topView.frame.height, width: self.view.frame.width, height: self.view.frame.height - topView.frame.height - tabBarController!.tabBar.frame.size.height))
         feedTableView.delegate = self
         feedTableView.dataSource = self
         feedTableView.backgroundColor = UIColor.podcastWhiteLight
```
Comment (anniexcheng) [inline Podcast/DiscoverViewController.swift]: I know this wasn't your change but don't need `UIColor` here

File: Podcast/DiscoverViewController.swift [hunk 5]
```diff
@@ -47,13 +47,15 @@ class DiscoverViewController: UIViewController, UITableViewDelegate, UITableView
         categoryCollectionView.reloadData()
         
         //tableview
-        feedTableView = UITableView(frame: CGRect(x: 0, y: topView.frame.height, width: self.view.frame.width, height: self.view.frame.height - topView.frame.height))
+        feedTableView = UITableView(frame: CGRect(x: 0, y: topView.frame.height, width: self.view.frame.width, height: self.view.frame.height - topView.frame.height - tabBarController!.tabBar.frame.size.height))
```
Comment (anniexcheng) [inline Podcast/DiscoverViewController.swift]: no `self`

**Comment by anniexcheng** [comment]
made some comments, otherwise looks good!

File: Podcast/DiscoverTableViewCell.swift [hunk 6]
```diff
@@ -41,6 +42,7 @@ class DiscoverTableViewCell: UITableViewCell {
     var moreButton: UIButton!
     var clickToPlayImageButton: UIButton!
     var seperator: UIView!
+    var isExpanded: Bool! = false
```
Comment (lucasderraugh) [inline Podcast/DiscoverTableViewCell.swift]: Shouldn't be implicitly unwrapped, also redundant type info.

File: Podcast/DiscoverTableViewCell.swift [hunk 7]
```diff
@@ -77,7 +80,7 @@ class DiscoverTableViewCell: UITableViewCell {
         frame.size.height = height
         backgroundColor = UIColor.white
         selectionStyle = .none
-        
+        self.isExpanded = false
```
Comment (lucasderraugh) [inline Podcast/DiscoverTableViewCell.swift]: No `self` needed

File: Podcast/DiscoverTableViewCell.swift [hunk 8]
```diff
@@ -25,10 +25,11 @@ class DiscoverTableViewCell: UITableViewCell {
     var clickToPlayButtonMinY: CGFloat = 13
     var padding: CGFloat = 10
     var episodeNameLabelHeight: CGFloat = 36
-    var iconButtonMinY: CGFloat = 109
+    var iconButtonMinYOffset: CGFloat = 12
     var seriesNameLabelMinY: CGFloat = 33
     var episodeDescriptionLabelMinY: CGFloat = 50
     var episodeNameLabelMinY: CGFloat = 4
+    var heightConstraint: NSLayoutConstraint?
```
Comment (lucasderraugh) [inline Podcast/DiscoverTableViewCell.swift]: Shouldn't be an optional.

File: Podcast/DiscoverTableViewCell.swift (hunk 1 above)
Comment (lucasderraugh) [inline Podcast/DiscoverTableViewCell.swift]: Should use `NSLayoutConstraint.activate(_:)` instead as NSView's `addConstraint(_:)` is soft deprecated.

File: Podcast/DiscoverViewController.swift [hunk 9]
```diff
@@ -47,13 +47,15 @@ class DiscoverViewController: UIViewController, UITableViewDelegate, UITableView
...
         feedTableView.delegate = self
         feedTableView.dataSource = self
         feedTableView.backgroundColor = UIColor.podcastWhiteLight
         feedTableView.separatorStyle = .none
         feedTableView.register(DiscoverTableViewCell.self, forCellReuseIdentifier: "DiscoverTableViewCellIdentifier")
         view.addSubview(feedTableView)
+        feedTableView.rowHeight = UITableViewAutomaticDimension
```
Comment (lucasderraugh) [inline Podcast/DiscoverViewController.swift]: Already the default value.

File: Podcast/DiscoverViewController.swift [hunk 10]
```diff
@@ -77,6 +79,22 @@ class DiscoverViewController: UIViewController, UITableViewDelegate, UITableView
...
+        
+        //episode static data
+        for i in 0..<4 {
+            let episode = Episode(id: i)
+            episode.descriptionText = "44 min • Warriors star Stephen Curry admits he’s getting annoyed by the stream of recent criticism, the possibility ... JHFJSHFGSL igsad asodhaisuhda asidgisag as hsiadgipasug siugdig asuigsi asigasidg asiugdiasgais asigdaisgd aisdgapisdg asidgaosig"
+            episode.smallArtworkImage = #imageLiteral(resourceName: "fillerImage")
+            episode.largeArtworkImage = #imageLiteral(resourceName: "fillerImage")
+            episode.series = series
+            episode.title = "Stephen Curry - EP10"
+            episode.dateCreated = Date.init()
```
Comment (lucasderraugh) [inline Podcast/DiscoverViewController.swift]: Should be `Date()`

Comment (lucasderraugh) [reply Podcast/DiscoverViewController.swift]: Probably should create an initializer with more arguments (can use some nice defaults as well). Most of these properties probably can and should become immutable.

File: Podcast/DiscoverViewController.swift [hunk 11]
```diff
@@ -107,32 +125,35 @@ class DiscoverViewController: UIViewController, UITableViewDelegate, UITableView
...
-    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat
-    {
-        return DiscoverTableViewCell().height
-    }
-    
-    
+
     func numberOfSections(in tableView: UITableView) -> Int {
         return 1
     }
```
Comment (lucasderraugh) [inline Podcast/DiscoverViewController.swift]: The default assumption is 1. Shouldn't implement it if it's not needed.

File: Podcast/Episode.swift [hunk 12]
```diff
@@ -14,6 +14,7 @@ class Episode: NSObject {
     var title: String = ""
     var series: Series?
     var dateCreated: Date?
+    var descriptionText: String?
```
Comment (lucasderraugh) [inline Podcast/Episode.swift]: If you add in the initializer with more arguments, attempt to make these non-optional.

**Pull request merged**